dashboard/src/actions/charts.ts

```diff
@@ -77,23 +64,32 @@ export function fetchCharts(
 ): ThunkAction<Promise<void>, IStoreState, null, ChartsAction> {
   return async dispatch => {
     dispatch(requestCharts());
-    const response = await httpGet(dispatch, url.api.charts.list(repo));
-    if (response) {
-      dispatch(receiveCharts(response));
+    try {
+      const charts = await Chart.fetchCharts(repo);
+      if (charts) {
+        dispatch(receiveCharts(charts));
+      }
+    } catch (e) {
+      dispatchError(dispatch, e);
     }
   };
 }
 
 export function fetchChartVersions(
   id: string,
-): ThunkAction<Promise<IChartVersion[]>, IStoreState, null, ChartsAction> {
+): ThunkAction<Promise<IChartVersion[] | null>, IStoreState, null, ChartsAction> {
   return async dispatch => {
     dispatch(requestCharts());
-    const response = await httpGet(dispatch, url.api.charts.listVersions(id));
-    if (response) {
-      dispatch(receiveChartVersions(response));
+    try {
+      const versions = await Chart.fetchChartVersions(id);
+      if (versions) {
+        dispatch(receiveChartVersions(versions));
+      }
+      return versions;
+    } catch (e) {
+      dispatchError(dispatch, e);
     }
-    return response;
+    return null;
```

Comment:
Is this required here? The `try` clause always returns (even if `!versions`), so it's just the `catch` clause which doesn't explicitly return?